### PR TITLE
chore: Add new features to RPC

### PR DESCRIPTION
**Pull request by ririxidev**
* added feature that instead of showing Apple Music logo, now it shows Song/Album artwork instead since Discord now allows to set external images. 
* deleted showing album in small image key, it's kinda useless when it already shows it in state. 
* made `Apple Music for macOS` -> `Apple Music`.
* added button to view song on AM website.
image

### CODE REVIEW

**Comment by NextFire** [comment]
Oh great! I was not aware they finally allowed external images.
Just on style, can I ask you to keep the indents set on 2 spaces? (note to myself: consider setting eslint)
I'll check the rest a bit later!

**Comment by ririxidev** [comment]
> Just on style, can I ask you to keep the indents set on 2 spaces? (note to myself: consider setting eslint)
You should start using https://prettier.io then, it's default spacing. eslint can follow prettier indents tbh but if you really need, just give thumbs up on this comm ty

**Comment by NextFire** [comment]
Thanks, I added a prettier configuration to the project

File: music-rpc.js [hunk 1]
```diff
@@ -44,6 +45,19 @@ function getProps() {
...
+  const { artist, album } = props;
+  const options = {
+    term: encodeURI(decodeURI(`${artist} ${album}`)),
+    media: 'music',
+    entity: 'album',
+    limit: 1,
+  };
+
+  const result = await itunesAPI.searchItunes(options);
+  return [result.results[0].artworkUrl100, result.results[0].collectionViewUrl];
```
Comment (NextFire) [inline music-rpc.js:58]: >   return [result.results[0].artworkUrl100, result.results[0].collectionViewUrl];
>                             ^
> 
> TypeError: Cannot read properties of undefined (reading 'artworkUrl100')

When no match is found (might happen with imported tracks)

Comment (ririxidev) [reply music-rpc.js]: fixed

File: music-rpc.js [hunk 2]
```diff
@@ -53,23 +72,32 @@ async function setActivity() {
...
           smallImageKey: state,
-          smallImageText:
-            `${state[0].toUpperCase() + state.slice(1)}『${props.name}』by ${
-              props.artist
-            }` + (props.year ? ` (${props.year})` : ''),
-        });
+          smallImageText: `${state[0].toUpperCase() + state.slice(1)}`,
+        };
+
+        if (!artwork[1]) {
```
Comment (NextFire) [inline music-rpc.js:91]:
```suggestion
        if (artwork[1] !== null) {
```
`!artwork[1]` was not enough...

Comment (ririxidev) [reply music-rpc.js]: Bruuuuuh js 😩

Comment (NextFire) [reply music-rpc.js]: Yeah I think I'll port the code to typescript soon...

Comment (ririxidev) [reply music-rpc.js]: Well I wanted to do it but no time rn

File: music-rpc.js [hunk 3]
```diff
@@ -44,6 +45,24 @@ function getProps() {
   });
 }
 
+async function searchSong(props) {
+  const { artist, album } = props;
+  const options = {
+    term: encodeURI(decodeURI(`${artist} ${album}`)),
```
Comment (NextFire) [inline music-rpc.js:51]: > encodeURI(decodeURI(...))

For which reasons do we need to do this ?

Comment (ririxidev) [reply music-rpc.js:51]: When there are Unicode characters like ł in polish when we will not do it, it will just throw error about characters

**Comment by NextFire** [comment]
Looks good now, I am going to merge this pr if there is no more changes. Thanks @ririxidev!

**Pull request merged**